compiler/src/dotty/tools/dotc/transform/VCElideAllocations.scala

```diff
@@ -23,13 +23,16 @@ class VCElideAllocations extends MiniPhase with IdentityDenotTransformer {
   override def runsAfter: Set[String] = Set(ElimErasedValueType.name)
 
   override def transformApply(tree: Apply)(using Context): Tree =
+    def hasUserDefinedEquals(tp: Type): Boolean =
+      val eql = defn.Any_equals.overridingSymbol(tp.typeSymbol.asClass)
+      eql.exists && !eql.is(Synthetic)
```

Comment:
Hmm, looks like vc-equals.scala is failing because we get a mixin forwarder `equals` in `D`, adding `butNot = Bridge` should fix that. Additionally, I think `overridingSymbol` only looks for symbols in the class itself since it calls `matchingDecl`, whereas we want `matchingMember` here.